api/config/v1alpha1/config_types.go

```diff
@@ -0,0 +1,25 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var _ runtime.Object = (*Config)(nil)
+
+// +kubebuilder:object:root=true
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Config is the Schema for the Apoxy Backplane configuration.
+type Config struct {
+	metav1.TypeMeta `json:",inline"`
+
+	Spec ConfigSpec `json:"spec,omitempty"`
```

Comment:
I think we don't need to have a Spec field here since it's not an actual API object so we don't need to follow this convention here and just drop all the settings in top level similar to this: https://github.com/kubernetes/kubelet/blob/c6be68c24f917323135cecc3ac03394d81a7f048/config/v1beta1/types.go#L89